pallets/fees/src/lib.rs

```diff
@@ -83,106 +102,82 @@ pub mod pallet {
 	#[pallet::genesis_build]
 	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
 		fn build(&self) {
-			for fee in self.initial_fees.iter() {
-				<Pallet<T>>::change_fee(fee.0, fee.1);
+			for (key, fee) in self.initial_fees.iter() {
+				<FeeBalances<T>>::insert(key, fee);
 			}
 		}
 	}
 
-	/// Stores the Fees associated with a Hash identifier
+	/// Stores the fee balances associated with a Hash identifier
 	#[pallet::storage]
 	#[pallet::getter(fn fee)]
-	pub(super) type Fees<T: Config> =
-		StorageMap<_, Blake2_256, T::Hash, Fee<T::Hash, BalanceOf<T>>>;
+	pub(super) type FeeBalances<T: Config> =
+		StorageMap<_, Blake2_256, T::FeeKey, BalanceOf<T>, ValueQuery, T::DefaultFeeValue>;
 
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
-		FeeChanged(T::Hash, BalanceOf<T>),
-	}
-
-	#[pallet::error]
-	pub enum Error<T> {
-		/// Fee associated to given key not found
-		FeeNotFoundForKey,
-		/// Invalid author of blocked fetched
-		InvalidAuthor,
+		FeeChanged(T::FeeKey, BalanceOf<T>),
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Set the given fee for the key
 		#[pallet::weight(<T as pallet::Config>::WeightInfo::set_fee())]
-		pub fn set_fee(
-			origin: OriginFor<T>,
-			key: T::Hash,
-			new_price: BalanceOf<T>,
-		) -> DispatchResult {
-			Self::can_change_fee(origin)?;
-			Self::change_fee(key, new_price);
-			Self::deposit_event(Event::FeeChanged(key, new_price));
+		pub fn set_fee(origin: OriginFor<T>, key: T::FeeKey, fee: BalanceOf<T>) -> DispatchResult {
+			T::FeeChangeOrigin::try_origin(origin)
```

Comment:
I think we can simplify this like:
```suggestion
			T::FeeChangeOrigin::ensure_origin(origin)?;
```

Reply:
I'm not sure to get the same behaviour with the `?`. Currently it is calling to `ensure_root` in case the `try_origin` (or ensure_origin) fails, so I still need to make the `or_else`.

Reply:
Yeah, you're right! Changed 😃